app/Model/CoExpirationPolicy.php

```diff
@@ -671,6 +704,58 @@ public function executePolicies($coId, $appShell=null) {
     
     return true;
   }
+
+    /**
+     * @param $co_id
+     * @param $cou_name
+     * @return false|mixed
+     */
+    public function retrieveCouEofId($co_id, $cou_name) {
+        // Currently i exclude all the EOF that refer to COU enrollment
+        $this->CoEnrollmentAttribute = ClassRegistry::init('CoEnrollmentAttribute');
+        $args = array();
+        $args['joins'][0]['table'] = 'co_enrollment_attribute_defaults';
+        $args['joins'][0]['alias'] = 'CoEnrollmentAttributeDefault';
+        $args['joins'][0]['type'] = 'INNER';
+        $args['joins'][0]['conditions'][0] = 'CoEnrollmentAttributeDefault.co_enrollment_attribute_id=CoEnrollmentAttribute.id';
+        $args['joins'][0]['conditions'][1] = 'CoEnrollmentAttribute.attribute iLIKE \'%cou%\'';
+        $args['joins'][0]['conditions'][2] = 'CoEnrollmentAttributeDefault.value ~ \'^[0-9]+$\'';
+        $args['joins'][0]['conditions'][3]['CoEnrollmentAttribute.deleted'] = false;
+        $args['joins'][0]['conditions'][4]['CoEnrollmentAttributeDefault.deleted'] = false;
+        $args['joins'][1]['table'] = 'cous';
+        $args['joins'][1]['alias'] = 'Cou';
+        $args['joins'][1]['type'] = 'INNER';
+        $args['joins'][1]['conditions'][0] = 'Cou.id=cast(CoEnrollmentAttributeDefault.value as integer)';
+        $args['joins'][1]['conditions'][1]['Cou.deleted'] = false;
+        $args['joins'][1]['conditions'][2]['Cou.name'] = $cou_name;
+        $args['joins'][1]['conditions'][2]['Cou.co_id'] = $co_id;
```

Comment:
Maybe the second $args['joins'][1]['conditions'][2] be -> $args['joins'][1]['conditions'][3]?

Reply:
Done